controllers/nodemaintenance_controller.go

```diff
@@ -419,6 +424,11 @@ func (r *NodeMaintenanceReconciler) onReconcileErrorWithRequeue(nm *nodemaintena
 		pendingList, _ := r.drainer.GetPodsForDeletion(nm.Spec.NodeName)
 		if pendingList != nil {
 			nm.Status.PendingPods = GetPodNameList(pendingList.Pods())
+			if nm.Status.EvictionPods != 0 {
+				drainProgress := float64((nm.Status.EvictionPods-len(nm.Status.PendingPods))*100) / float64(nm.Status.EvictionPods) // Calculate progress
+				nm.Status.DrainProgress = math.Round(drainProgress*100) / 100  // Take only twp decimal numbers
+			}
+			nm.Status.ElapsedTime = time.Since(nm.ObjectMeta.GetCreationTimestamp().UTC()) / time.Duration(math.Pow10(9)) // Drain's elapsed time from [ns] to [s]
```

Comment:
I have been tweaking with casting few times when calculating the two fields :(

- I have been using it solely to cast/transform _Time_ to _time.Time_ after getting the following error `cannot use nm.ObjectMeta.GetCreationTimestamp() (value of type "k8s.io/apimachinery/pkg/apis/meta/v1".Time) as time.Time value in argument to time.Since`.

- > Whoever is going to display the value later on will chose the value for the desired unit by calling the appropriate method like e.g. Seconds().

It is easier and make sense to not forcefully display it in one way and I didn't know metav1.Duration is human readable :)

-  Your input did simplify the code and even make it shorter.